packages/actor-rdf-resolve-quad-pattern-federated/lib/FederatedQuadSource.ts

```diff
@@ -83,6 +89,56 @@ export class FederatedQuadSource implements ILazyQuadSource {
     return term;
   }
 
+  /**
+   * If the given term is a blank node, return a deterministic named node for it
+   * based on the source id and the blank node value.
+   * @param term Any RDF term.
+   * @param sourceId A numerical source identifier.
+   * @return If the given term was a blank node, this will return a skolemized named node, otherwise the original term.
+   */
+  public static skolemizeTerm(term: RDF.Term, sourceId: number): RDF.Term | BlankNodeSkolemizable {
+    if (term.termType === 'BlankNode') {
+      const skolemized = `urn:comunica_skolem:source_${sourceId}:${term.value}`;
+      return new BlankNodeSkolemizable(skolemized, DataFactory.namedNode(skolemized));
```

Comment:
If this is consistently the case, should we just define `value` as a getter in `BlankNodeSkolemizable` that returns `this.skolemized.value`? Or are you relying on polymorphism for making things fast in V8 (valid argument)?

Reply:
Alternative: wondering if we want `value` to be something more human-accessible (while still keeping in mind unicity). Like `_:bc_1_1` (for Blanknode Comunica sourceId value) or so.